iib/web/models.py

```diff
@@ -1943,3 +1949,133 @@ def get_mutable_keys(self) -> Set[str]:
+            raise ValidationError('The "binary_image" value must be a non-empty string')
+
+        # Verify that `overwrite_from_index` is the correct type
+        overwrite = request_kwargs.pop('overwrite_from_index', False)
+        if not isinstance(overwrite, bool):
+            raise ValidationError('The "overwrite_from_index" parameter must be a boolean')
+
+        # Verify that `overwrite_from_index_token` is the correct type
+        overwrite_token = request_kwargs.pop('overwrite_from_index_token', None)
+        if overwrite_token:
+            if not isinstance(overwrite_token, str):
+                raise ValidationError('The "overwrite_from_index_token" parameter must be a string')
+            if overwrite_token and not overwrite:
+                raise ValidationError(
+                    'The "overwrite_from_index" parameter is required when'
+                    ' the "overwrite_from_index_token" parameter is used'
+                )
+
+        # Remove attributes that are not stored in the database
+        request_kwargs.pop('overwrite_from_index_token', None)
+
+        # Validate add_arches are correctly provided
+        add_arches = request_kwargs.pop('add_arches', [])
+        Architecture.validate_architecture_json(add_arches)
+
+        # Verify the user is authorized to use overwrite_from_index
+        # current_user.is_authenticated is only ever False when auth is disabled
+        if current_user.is_authenticated:
+            if overwrite and not overwrite_token:
+                raise Forbidden(
+                    'You must set "overwrite_from_index_token" to use "overwrite_from_index"'
+                )
+
+        # Add the request to a new batch
+        batch = Batch()
+        db.session.add(batch)
+        request_kwargs['batch'] = batch
+
+        request = cls(**request_kwargs)
+        request.add_state('in_progress', 'The request was initiated')
+        return request
+
+    def to_json(self, verbose: Optional[bool] = True) -> FbcOperationRequestResponse:
+        """
+        Provide the JSON representation of a "fbc-operation" build request.
+
+        :param bool verbose: determines if the JSON output should be verbose
+        :return: a dictionary representing the JSON of the build request
+        :rtype: dict
+        """
+        # cast to result type, super-type returns Union
+        rv = cast(FbcOperationRequestResponse, super().to_json(verbose=verbose))
+        rv['fbc_fragment'] = self.fbc_fragment.pull_specification
+        rv['fbc_fragment_resolved'] = getattr(
+            self.fbc_fragment_resolved, 'pull_specification', None
+        )
+
+        return rv
+
+    def get_mutable_keys(self) -> Set[str]:
+        """
+        Return the set of keys representing the attributes that can be modified.
+
+        :return: a set of key names
+        :rtype: set
+        """
+        rv = super().get_mutable_keys()
+        rv.add('fbc_fragment_resolved')
```

Comment:
You probably also need `rv.update(self.get_index_image_mutable_keys())`? Am I missing something?

Reply:
+1 this is needed.
@lipoja this was the change I had to add to ensure update_request is working

Reply:
Thank you - added